freqtrade/optimize/backtesting.py

```diff
@@ -1226,13 +1225,11 @@ def backtest_one_strategy(self, strat: IStrategy, data: Dict[str, DataFrame],
         self._set_strategy(strat)
 
         # Use max_open_trades in backtesting, except --disable-max-market-positions is set
-        if self.config.get('use_max_market_positions', True):
-            # Must come from strategy config, as the strategy may modify this setting.
-            max_open_trades = self.strategy.config['max_open_trades']
-        else:
+        if not self.config.get('use_max_market_positions', True):
             logger.info(
                 'Ignoring max_open_trades (--disable-max-market-positions was used) ...')
-            max_open_trades = 0
+            self.strategy.max_open_trades = -1
+            self.config.update({'max_open_trades': float('inf')})
```

Comment:
for my understanding mostly ...
why are we assigning different values to strategy and config?

while `strategy.max_open_trades` won't be used for the most part - won't this create a problem should that value be used? (it's clearly unexpected to have the 2 values diverge).

Reply:
There is a little ambiguity with typing since the possible values for `max_open_trades` in the config are positive integers and float('inf') (which is a float), while the ones in the strategy are integers starting from -1 (so only `int`). 

I am going to check if it is better to have also the `max_open_trades` from the config being only `int`, with -1 meaning infinite, so that the two values will always match. What do you think about it? 

And also, yes, it is true, all of the reading of the `max_open_trades` parameter are done from the config, `max_open_trades` in the strategy is only used to sync with the config when loading the strategy, so I can avoid updating the strategy.max_open_trades while hyperopting / backtesting, but I thought it would have been better to have them synced (even if it is quite hardcoded).

Reply:
i think setting it in both is better - if it's available, people will use it in the strategy for "whatever" (like `stake_amount / self.max_open_trades` (or whatever).
Results would differ if it's set "wrongly" during hyperopt.